chromium_src/components/omnibox/browser/omnibox_client.h

```diff
@@ -0,0 +1,14 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_COMPONENTS_OMNIBOX_BROWSER_OMNIBOX_CLIENT_H_
+#define BRAVE_CHROMIUM_SRC_COMPONENTS_OMNIBOX_BROWSER_OMNIBOX_CLIENT_H_
+
+#define BRAVE_OMNIBOX_CLIENT_H \
+  virtual bool IsAutocompleteEnabled() const;
```

Comment:
There's no point in making this a virtual method. We know the omnibox client is BraveOmniboxClient so just cast it in BraveOmniboxController

Reply:
@bridiver  I think this causes layer violation because `BraveOmniboxController` is in component layer but `BraveOmniboxClient` is in chrome layer. WDYT?

Reply:
then make an abstract `BraveOmniboxClient` with `IsAutocompleteEnabled` in components and have `BraveOmniboxClientImpl` in the chrome layer

Reply:
then you can cast to `BraveOmniboxClient` without a layering violation

Reply:
@bridiver Thanks for tips!
Updated and this PR doesn't have new patch files :) PTAL.